### PR TITLE
Fixed the dependecy of FrameSettings (MSAA, ClearGBuffer, DepthPrepassWithDeferred) (case 1277620).

### PR DESCRIPTION
**Pull request by anisunity**
https://fogbugz.unity3d.com/f/cases/1277620/
As reported, on the custome frame settings, the set of frame settings that can be enabled was wrong. This PR fixes the issue for MSAA, clearGbuffer and DepthPrepassWithDeferred.

Testing status
I tested all the combination of deferred forward and both on both the asset and the custom frame settings, the behavior made sense.

### CODE REVIEW

**Review by JMargevics — APPROVED** [review]
Straightforward fix, Anis wrote what tested. (auto)Approving as is ✅

**Review by RSlysz — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs [hunk 1]
```diff
@@ -171,12 +171,13 @@ static void Drawer_SectionRenderingSettings(SerializedFrameSettings serialized,
...
+
             bool frameSettingsOverrideToForward = serialized.GetOverrides(FrameSettingsField.LitShaderMode) && serialized.litShaderMode == LitShaderMode.Forward;
             bool frameSettingsOverrideToDeferred = serialized.GetOverrides(FrameSettingsField.LitShaderMode) && serialized.litShaderMode == LitShaderMode.Deferred;
             bool defaultForwardUsed = !serialized.GetOverrides(FrameSettingsField.LitShaderMode) && defaultShaderLitMode == LitShaderMode.Forward;
-            bool defaultDefferedUsed = !serialized.GetOverrides(FrameSettingsField.LitShaderMode) && defaultShaderLitMode == LitShaderMode.Deferred;
-            bool msaaEnablable = hdrpAssetSupportForward && hdrpSettings.supportMSAA && (frameSettingsOverrideToForward || defaultForwardUsed);
-            bool depthPrepassEnablable = hdrpAssetSupportDeferred && (defaultDefferedUsed || frameSettingsOverrideToDeferred);
+            bool msaaEnablable = (hdrpAssetSupportForward && hdrpSettings.supportMSAA && (frameSettingsOverrideToForward || defaultForwardUsed)) || (hdrpAssetIsForward && hdrpSettings.supportMSAA);
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:180]: typo msaaEnablable

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs [hunk 2]
```diff
@@ -171,12 +171,13 @@ static void Drawer_SectionRenderingSettings(SerializedFrameSettings serialized,
 
             bool hdrpAssetSupportForward = hdrpSettings.supportedLitShaderMode != RenderPipelineSettings.SupportedLitShaderMode.DeferredOnly;
             bool hdrpAssetSupportDeferred = hdrpSettings.supportedLitShaderMode != RenderPipelineSettings.SupportedLitShaderMode.ForwardOnly;
+            bool hdrpAssetIsForward = hdrpSettings.supportedLitShaderMode == RenderPipelineSettings.SupportedLitShaderMode.ForwardOnly;
+            bool hdrpAssetIsDeffered = hdrpSettings.supportedLitShaderMode == RenderPipelineSettings.SupportedLitShaderMode.DeferredOnly;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:175]: type deffered => deferred

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs [hunk 3]
```diff
@@ -194,6 +195,9 @@ static void Drawer_SectionRenderingSettings(SerializedFrameSettings serialized,
                             throw new System.ArgumentOutOfRangeException("Unknown ShaderLitMode");
                     }
                 });
+
+            bool defaultDefferedUsed = !serialized.GetOverrides(FrameSettingsField.LitShaderMode) && defaultShaderLitMode == LitShaderMode.Deferred;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:199]: typo deffered

**Review by JulienIgnace-Unity — CHANGES_REQUESTED** [review]
Just a few typos, otherwise looks good to me.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs (hunk 1 above)
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:180]: Not a typo https://en.wiktionary.org/wiki/enablable, tho it is was written before @JulienIgnace-Unity

**Comment by anisunity** [comment]
Fixed the other typos

**Review by JulienIgnace-Unity — APPROVED** [review]

**Pull request merged**